### PR TITLE
Improvement of tests

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
This PR is a work-in-progress for improvement of UpTrain tests.

- Current tests do not really test individual components extensively and are code from the example files.
- Smaller unit tests can be used to cover different scenarios, edge cases and error conditions.
- Tests should be more modular, readable and maintainable for future contributors, developers and users.
- Breaking down tests into smaller parts will help automate them via workflows and github actions if/when integrated.

Edit: Updates to tests will be spread across multiple PRs to make reviewing easier as discussed. For now, this PR contains the following changes:
- Documentation and explanation of Concept Drift Adwin test
- Change of Concept Drift DDM test to use synthetic data instead of code from the examples
- Addition of tests for `uptrain.core.classes.measurables.accuracy`

### CODE REVIEW

File: uptrain/tests/test_measurables_accuracy.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pytest
+
+from uptrain.core.classes.measurables import (
```
Comment (vipgupta) [inline uptrain/tests/test_measurables_accuracy.py:4]: Hey @a-r-r-o-w , do we need this file now? ADWIN already tests for MAE. I think we can also add MAPE there? And DDM tests for AccuracyMeasurable

Comment (a-r-r-o-w) [reply uptrain/tests/test_measurables_accuracy.py]: Yes, we do not require this file since it does not test the entire pipeline in itself, which is our goal when writing these tests. I've removed it and made other discussed changes.

The only thing that remains is fixing the plot bug which can be seen on the dashboard when running the current tests in this PR.

**Review by vipgupta — APPROVED** [review]
